lib/srv/db/cloud/resource_checker_url_aws.go

```diff
@@ -0,0 +1,238 @@
+	"github.com/gravitational/teleport/lib/services"
+)
+
+func (c *urlChecker) checkAWS(describeCheck, basicEndpointCheck checkDatabaseFunc) checkDatabaseFunc {
+	return func(ctx context.Context, database types.Database) error {
+		err := describeCheck(ctx, database)
+
+		// Database Service may not have enough permissions to permform the
+		// describes. Log a warning and permform a basic endpoint validation
+		// instead.
+		if trace.IsAccessDenied(err) {
+			c.logAWSAccessDeniedError(database, err)
+
+			if err := basicEndpointCheck(ctx, database); err != nil {
+				return trace.Wrap(err)
+			}
+			c.log.Debugf("Database %v URL validated by basic endpoint check.", database.GetName())
+			return nil
+		}
+
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		c.log.Debugf("Database %v URL validated by describe check.", database.GetName())
+		return nil
+	}
+}
+
+func (c *urlChecker) logAWSAccessDeniedError(database types.Database, accessDeinedError error) {
+	c.warnAWSOnce.Do(func() {
+		// TODO(greedy52) add links to doc.
+		c.log.Warn("No permissions to describe AWS resource metadata that is needed for validating databases created by Discovery Service. Basic AWS endpoint validation will be performed instead. For best security, please provide the Database Service with the proper IAM permissions. Enable --debug mode to see details on which databases require more IAM permissions. See Database Access documentation for more details.")
+	})
+
+	c.log.Debugf("No permissions to describe %q: %v.", database, accessDeinedError)
+}
+
+func (c *urlChecker) checkRDS(ctx context.Context, database types.Database) error {
+	meta := database.GetAWS()
+	rdsClient, err := c.clients.GetAWSRDSClient(ctx, meta.Region, cloud.WithAssumeRoleFromAWSMeta(meta))
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	if meta.RDS.ClusterID != "" {
+		return trace.Wrap(c.checkRDSCluster(ctx, database, rdsClient, meta.RDS.ClusterID))
+	}
+	return trace.Wrap(c.checkRDSInstance(ctx, database, rdsClient, meta.RDS.InstanceID))
+}
+func (c *urlChecker) checkRDSInstance(ctx context.Context, database types.Database, rdsClient rdsiface.RDSAPI, instanceID string) error {
+	rdsInstance, err := describeRDSInstance(ctx, rdsClient, instanceID)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	if rdsInstance.Endpoint == nil {
+		return trace.BadParameter("empty endpoint")
+	}
+	return trace.Wrap(requireDatabaseAddressPort(database, rdsInstance.Endpoint.Address, rdsInstance.Endpoint.Port))
+}
+func (c *urlChecker) checkRDSCluster(ctx context.Context, database types.Database, rdsClient rdsiface.RDSAPI, clusterID string) error {
+	rdsCluster, err := describeRDSCluster(ctx, rdsClient, clusterID)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	databases, err := services.NewDatabasesFromRDSCluster(rdsCluster)
+	if err != nil {
+		c.log.Warnf("Could not convert RDS cluster %q to database resources: %v.",
+			aws.StringValue(rdsCluster.DBClusterIdentifier), err)
+	}
```

Comment:
Can we return a nil immediately without calling the `requireContainsDatabaseURLAndType` function with `nil` object ? 
```suggestion
	if err != nil {
		c.log.Warnf("Could not convert RDS cluster %q to database resources: %v.",
			aws.StringValue(rdsCluster.DBClusterIdentifier), err)
	}
	return nil
```

Reply:
updated:
```
    databases, err := services.NewDatabasesFromRDSCluster(rdsCluster)                                
    if err != nil {                                                                                                                         
        c.log.Warnf("Could not convert RDS cluster %q to database resources: %v.",                                
            aws.StringValue(rdsCluster.DBClusterIdentifier), err)                                
                                     
        // services.NewDatabasesFromRDSCluster maybe partially successful.                                
        if len(databases) == 0 {                                                                                                   
            return nil                                                                                 
        }                                                                                     
    } 
```